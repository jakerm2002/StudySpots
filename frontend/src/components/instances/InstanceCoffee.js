import axios from "axios";
import styles from '../instance_components/InstanceTemplate.module.css'
import React, { useState, useEffect } from "react";
import { useParams } from "react-router-dom";
import MapComponent from "../instance_components/MapComponent";
import NearbyLibrary from '../nearby/NearbyLibrary.js';
import NearbyUniversity from '../nearby/NearbyUniversity.js';
import WeatherWidget from "../instance_components/WeatherWidget";
import { Divider } from "@mui/material";
import { Button, Carousel, Container, Figure, Row, Col, Accordion, Card } from "react-bootstrap";
import TravelTime from "../instance_components/TravelTime";


const InstanceCoffee = () => {
  const { businessID } = useParams();
  const [isLoading, setIsLoading] = useState(true);
  const [data, setData] = useState([]);

  useEffect(() => {
    axios.get('https://api.studyspots.me/coffeeshops/' + businessID).then(response => {
      console.log("response", response.data);
      setData(response.data);
      console.log(data)
      setIsLoading(false);
    });
  }, []);

  return (
    <>
      {!isLoading && (
        <div className={`text ${styles.general} topSpacing`}>
          <h1>{data.name}</h1>
          <WeatherWidget latitude={data.latitude} longitude={data.longitude} />
          <Divider className={styles.divider}>â˜•</Divider>
          <Container className={styles.spacing}>
            <Carousel>
              {data.image_url !== ""
                ? <Carousel.Item>
                  <Figure.Image
                    className={`d-block w-50 ${styles.image}`}
                    src={data.image_url}
                    alt={data.name}
                  />
                  <Figure.Caption className={styles.caption}>
                    Image Source: {data.image_url}
                  </Figure.Caption>
                </Carousel.Item>
                : <Carousel.Item>
                  <img
                    className={`d-block w-50 ${styles.image}`}
                    src="https://www.escapeauthority.com/wp-content/uploads/2116/11/No-image-found.jpg"
                    alt="Unable to display coffee shop"
                  />
                </Carousel.Item>
              }
              {data.image_2_url !== "" && (
                <Carousel.Item>
                  <Figure.Image
                    className={`d-block w-50 ${styles.image}`}
                    src={data.image_2_url}
                    alt={data.name}
                  />
                  <Figure.Caption className={styles.caption}>
                    Image Source: {data.image_2_url}
                  </Figure.Caption>
                </Carousel.Item>
              )}
              {data.image_3_url !== "" && (
                <Carousel.Item>
                  <Figure.Image
                    className={`d-block w-50 ${styles.image}`}
                    src={data.image_3_url}
                    alt={data.name}
                  />
                  <Figure.Caption className={styles.caption}>
                    Image Source: {data.image_3_url}
                  </Figure.Caption>
                </Carousel.Item>
              )}
            </Carousel>
          </Container>
          <Container className={`cards ${styles.spacing} ${styles.styleCard}`}>
            <Row>
              <Col>
                <div>
                  <b>Address:</b><br />
                  {data.address1}<br />
                  {data.city}, {data.state} {data.zipcode}
                </div>
                <div>
                  <b>Phone Number:</b> {data.phone}
                </div>
                <div>
                  <b>Price:</b> {data.price}
                </div>
              </Col>
              <Col>
                <div>
                  <b>Hours (in CT):</b> <br />
                  <div className={styles.hours}>
                    {data.formatted_hours}
                  </div>
                </div>
              </Col>
            </Row>
            <Button
              className={styles.spacing}
              variant="dark"
              onClick={() =>
                navigator.clipboard.writeText(
                  data.name + '\n' + data.address1 + '\n' + data.city + ' ' + data.state + ' ' + data.zipcode + '\n' + data.phone
                )
              }
            >
              Copy Information
            </Button>
          </Container>
          <Container className={`cards ${styles.spacing} ${styles.styleCard}`}>
            <div>
              <b>Overall Rating:</b> {data.rating.toFixed(1)}
            </div>
            <div>
              <b>Number of Reviews:</b> {data.review_count}
            </div>
            <Row>
              <Col>
                <Card className={`cards`}>
                  {data.review_1_author !== "N/A"
                    ? <Card.Body><Card.Title>
                      {data.review_1_rating + "/5 stars"}
                    </Card.Title>
                      <Card.Subtitle>
                        {data.review_1_author}
                      </Card.Subtitle>
                      <Card.Text>
                        {data.review_1_text}
                      </Card.Text>
                    </Card.Body>
                    : <Card.Title>
                      {"No Rating Available"}
                    </Card.Title>
                  }
                </Card>
              </Col>
              <Col>
                <Card className={`cards`}>
                  {data.review_2_author !== "N/A"
                    ? <Card.Body><Card.Title>
                      {data.review_2_rating + "/5 stars"}
                    </Card.Title>
                      <Card.Subtitle>
                        {data.review_2_author}
                      </Card.Subtitle>
                      <Card.Text>
                        {data.review_2_text}
                      </Card.Text>
                    </Card.Body>
                    : <Card.Title>
                      {"No Rating Available"}
                    </Card.Title>
                  }
                </Card>
              </Col>
              <Col>
                <Card className={`cards`}>
                  {data.review_3_author !== "N/A"
                    ? <Card.Body><Card.Title>
                      {data.review_3_rating + "/5 stars"}
                    </Card.Title>
                      <Card.Subtitle>
                        {data.review_3_author}
                      </Card.Subtitle>
                      <Card.Text>
                        {data.review_3_text}
                      </Card.Text>
                    </Card.Body>
                    : <Card.Title>
                      {"No Rating Available"}
                    </Card.Title>
                  }
                </Card>
              </Col>
            </Row>
          </Container>
          <Container className={`cards ${styles.spacing} ${styles.styleCard}`}>
            <h4>Map</h4>
            <MapComponent name={data.name} address={data.address1} latitude={data.latitude} longitude={data.longitude} />
            <TravelTime instanceLatitude={data.latitude} instanceLongitude={data.longitude} />
          </Container>
          <Container className={styles.spacing}>
            <Row>
              <Col className={`cards ${styles.spacing} ${styles.styleCard} ${styles.spacing_side}`}>
                <NearbyUniversity latitude={data.latitude} longitude={data.longitude} />
              </Col>
              <Col className={`cards ${styles.spacing} ${styles.styleCard} ${styles.spacing_side}`}>
                <NearbyLibrary latitude={data.latitude} longitude={data.longitude} />
              </Col>
            </Row>
          </Container>
        </div>
      )}
    </>
  );
}

export default InstanceCoffee;