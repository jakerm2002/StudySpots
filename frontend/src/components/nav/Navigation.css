#light .navBar {
    background-color: #78330c;
}

#dark .navBar {
    background-color: #e7dbcf;
}

#light .Nav-link {
    color: white;
  }

  #dark .Nav-link {
    color: #78330c;
  }