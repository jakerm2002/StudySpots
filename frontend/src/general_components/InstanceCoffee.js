import React, { useState, useEffect } from "react";
import { useParams, Link, BrowserRouter as Router, Route } from "react-router-dom";


import Figure from 'react-bootstrap/Figure';
import Container from 'react-bootstrap/Card';
import Accordion from 'react-bootstrap/Accordion';
import { Row } from "react-bootstrap";
import { Col } from "react-bootstrap";
import './InstanceTemplate.css';
import Splash from '../pages/Splash'
import MapComponent from "./MapComponent";


const InstanceCoffee = () => {
    console.log("SUPPPPP");

    const { businessID } = useParams();
      console.log(businessID);
      const [isLoading, setIsLoading] = useState(true);
      const [data, setData] = useState();
    
      useEffect(() => {
        fetch(`../instance_apis/${businessID}.json`, {
            headers: {
                'Content-Type': 'application/json',
                'Accept': 'application/json'
            }
        })
          .then((res) => res.json())
          .then((response) => {
            setData(response);
            setIsLoading(false);
          })
          .catch((error) => console.log(error));
      }, [businessID]);

    return (
        <>
          {!isLoading && (
            <div className="bennu-coffee-austin">
             	<Container className="header" >
             		<Row>
             			<Col className="image">
         				<Figure>
             					<Figure.Image src={data.image_url} />
             				</Figure>
             			</Col>
             			<Col className="stats">
                            <div className="stat">{data.name}</div>
                            <div className="stat">{data.display_phone}</div>
                            <div className="stat">{data.review_count} reviews</div>
                            <div className="stat">{data.location.address1}</div>
                            <div className="stat">{data.location.city} {data.location.state} {data.location.zip_code}</div>
                            <div className="stat">Price: {data.price}</div>
                            <div className="stat">Rating: {data.rating.toFixed(1)}</div>
                            <div className="stat">Website: <a href={data.url}>{data.url}</a></div>
             			</Col>
                        <MapComponent name={data.name} address={data.location.address1} latitude={data.coordinates.latitude} longitude={data.coordinates.longitude}/>
             		</Row>
             	</Container>
             	<Accordion className="info">
             	<Accordion.Item eventKey="0">
             		<Accordion.Header>Title</Accordion.Header>
             		    {/* <Accordion.Body className="body"> {body_text}
             		</Accordion.Body> */}
                    {/* supposed to be description here */}
             	</Accordion.Item>
             	</Accordion>
             	</div>
          )}
        </>
    );
}

export default InstanceCoffee;