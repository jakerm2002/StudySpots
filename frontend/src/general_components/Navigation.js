import { Navbar, Nav, Container } from 'react-bootstrap';

const Navigation = () => {
    return (
        <>
            <Navbar bg='dark' variant='dark'>
                <Container>
                    <Nav>
                        <Nav.Link href='/'>Home</Nav.Link>
                        <Nav.Link href='/CoffeeShops'>Coffee Shops</Nav.Link>
                        <Nav.Link href='/Libraries'>Libraries</Nav.Link>
                        <Nav.Link href='/Universities'>Universities</Nav.Link>
                        <Nav.Link href='/About'>About</Nav.Link>
                    </Nav>
                </Container>
            </Navbar>
        </>
    )
}

export default Navigation