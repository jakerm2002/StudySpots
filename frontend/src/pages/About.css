.profiles {
    justify-content: center;
}

.team {
    margin-left: auto;
    margin-right: auto;
}

.profileCard {
    margin: 25px;
}

.info {
    margin-top: 40px;
    margin-bottom: 40px;
}

.techInfo {
    margin-top: 40px;
    margin-bottom: 40px;
}

.centerTable {
    margin-left:auto;
    margin-right: auto;
    width: 75%;
}

table {
    overflow: auto;
    display: block;
    table-layout: auto;
}