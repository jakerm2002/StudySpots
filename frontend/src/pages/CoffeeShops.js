<<<<<<< HEAD
import ModelPageTemplate from '../general_components/ModelPageTemplate'

const CoffeeShops = () => {
    return (
        <div className="wrapper-all">
            <h1>Coffee Shops</h1>
            <ModelPageTemplate/>
        </div>
    )   
=======
const CoffeeShops = () => {
>>>>>>> 87ce49063973570e7951cdc95df6709aa404cb1c
}

export default CoffeeShops;