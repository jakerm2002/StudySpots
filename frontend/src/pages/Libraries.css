.modelCardContainer {
    justify-content: right;
    padding: 16px;
}

.modelCard {
    width: 20rem;
}

.image {
    margin-top: 2rem;
}

.map {
    margin-top: 2rem;
}