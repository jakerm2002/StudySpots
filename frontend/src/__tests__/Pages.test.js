import React from "react"
import About from "../pages/About"
import CoffeeShops from "../pages/CoffeeShops"
import Libraries from "../pages/Libraries"
import Universities from "../pages/Universities"
import InstanceCoffee from "../general_components/InstanceCoffee"
import Navigation from "../general_components/Navigation"
import Paginate from "../general_components/Pagination"
import renderer, { act } from 'react-test-renderer'

it("Renders About page", async () => {
    const tree = renderer.create(<About/>).toJSON();
    expect(tree).toMatchSnapshot();
});

it("Renders Coffee Shops page", async () => {
    const tree = renderer.create(<CoffeeShops/>).toJSON();
    expect(tree).toMatchSnapshot();
});

it("Renders Libraries page", async () => {
    const tree = renderer.create(<Libraries/>).toJSON();
    expect(tree).toMatchSnapshot();
});

it("Renders Universities page", async () => {
    const tree = renderer.create(<Universities/>).toJSON();
    expect(tree).toMatchSnapshot();
});

it("Renders Navigation bar", async () => {
    const tree = renderer.create(<Navigation/>).toJSON();
    expect(tree).toMatchSnapshot();
});

it("Renders Pagination Component", async () => {
    const tree = renderer.create(<Paginate/>).toJSON();
    expect(tree).toMatchSnapshot();
});