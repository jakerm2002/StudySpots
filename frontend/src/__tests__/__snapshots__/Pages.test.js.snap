// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders About page 1`] = `
Array [
  <div
    className="info"
  >
    <h1>
      StudySpots
    </h1>
    <h3>
      Mission
    </h3>
    <div
      className="center"
    >
      <p>
        StudySpots focuses on helping college students explore the areas around their campus while finding less busy places to work. StudySpots aims at helping college students quickly find locations nearby them and locations that meet all of the requirements for their standards to help them quickly get their work done.
      </p>
    </div>
    <h3>
      Discovery
    </h3>
    <div
      className="center"
    >
      <p>
        When creating these connections between different libraries, universities, and coffee shops, we noticed...
      </p>
    </div>
  </div>,
  <div
    className="teamInfo"
  >
    <h1>
      Meet the Team
    </h1>
    <h3>
      Members
    </h3>
    <div />
  </div>,
  <div
    className="techInfo"
  >
    <h1>
      Technology Used
    </h1>
    <h3>
      Tools
    </h3>
    <div
      className="center"
    >
      <table
        className="aboutTable table"
      >
        <tbody>
          <tr>
            <td>
              <div
                className="card"
                style={
                  Object {
                    "width": "15rem",
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    className="card-img-top"
                    src="react_logo.png"
                  />
                  <a
                    href="https://reactjs.org/"
                  >
                    <div
                      className="card-title h5"
                    >
                      React
                    </div>
                  </a>
                  <p
                    className="card-text"
                  >
                    We used React to create the overall website UI and design. 
                  </p>
                </div>
              </div>
            </td>
            <td>
              <div
                className="card"
                style={
                  Object {
                    "width": "15rem",
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    className="card-img-top"
                    src="react_router_logo.png"
                  />
                  <a
                    href="https://reactrouter.com/en/main"
                  >
                    <div
                      className="card-title h5"
                    >
                      React Router
                    </div>
                  </a>
                  <p
                    className="card-text"
                  >
                    We used React Router in order to create connections and allow navigation between the different pages.
                  </p>
                </div>
              </div>
            </td>
            <td>
              <div
                className="card"
                style={
                  Object {
                    "width": "15rem",
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    className="card-img-top"
                    src="react_bootstrap.png"
                  />
                  <a
                    href="https://react-bootstrap.github.io/"
                  >
                    <div
                      className="card-title h5"
                    >
                      React Bootstrap
                    </div>
                  </a>
                  <p
                    className="card-text"
                  >
                    We used React Bootstrap since it already included some basic components that we could use.
                  </p>
                </div>
              </div>
            </td>
            <td>
              <div
                className="card"
                style={
                  Object {
                    "width": "15rem",
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    className="card-img-top"
                    src="aws.png"
                  />
                  <a
                    href="https://aws.amazon.com/amplify/"
                  >
                    <div
                      className="card-title h5"
                    >
                      AWS Amplify
                    </div>
                  </a>
                  <p
                    className="card-text"
                  >
                    We used AWS Amplify to host the website.
                  </p>
                </div>
              </div>
            </td>
            <td>
              <div
                className="card"
                style={
                  Object {
                    "width": "15rem",
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    className="card-img-top"
                    src="docker.png"
                  />
                  <a
                    href="https://www.docker.com/"
                  >
                    <div
                      className="card-title h5"
                    >
                      Docker
                    </div>
                  </a>
                  <p
                    className="card-text"
                  >
                    We used Docker to be able to create the environment and all the installations necessary for development.
                  </p>
                </div>
              </div>
            </td>
            <td>
              <div
                className="card"
                style={
                  Object {
                    "width": "15rem",
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    className="card-img-top"
                    src="gitlab_logo.png"
                  />
                  <a
                    href="https://gitlab.com/"
                  >
                    <div
                      className="card-title h5"
                    >
                      GitLab
                    </div>
                  </a>
                  <p
                    className="card-text"
                  >
                    We used GitLab for source control, collaboration, and to keep track of issues.
                  </p>
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <h3>
      APIs Scraped
    </h3>
    <div
      className="center"
    >
      <table
        className="aboutTable table"
      >
        <tbody>
          <tr>
            <td>
              <div
                className="card"
                style={
                  Object {
                    "width": "15rem",
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    className="card-img-top"
                    src="gitlab_logo.png"
                  />
                  <a
                    href="https://docs.gitlab.com/ee/api/"
                  >
                    <div
                      className="card-title h5"
                    >
                      GitLab API
                    </div>
                  </a>
                  <p
                    className="card-text"
                  >
                    We used the GitLab API to fetch information about the number of commits and issues on the repo.
                  </p>
                </div>
              </div>
            </td>
            <td>
              <div
                className="card"
                style={
                  Object {
                    "width": "15rem",
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    className="card-img-top"
                    src="google_maps.png"
                  />
                  <a
                    href="https://developers.google.com/maps"
                  >
                    <div
                      className="card-title h5"
                    >
                      Google Maps API
                    </div>
                  </a>
                  <p
                    className="card-text"
                  >
                    We used the Google Maps API to fetch information for each of the library instances.
                  </p>
                </div>
              </div>
            </td>
            <td>
              <div
                className="card"
                style={
                  Object {
                    "width": "15rem",
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    className="card-img-top"
                    src="yelp_logo.png"
                  />
                  <a
                    href="https://www.yelp.com/developers"
                  >
                    <div
                      className="card-title h5"
                    >
                      Yelp API
                    </div>
                  </a>
                  <p
                    className="card-text"
                  >
                    We used the Yelp API to display information for each of the coffee shops.
                  </p>
                </div>
              </div>
            </td>
            <td>
              <div
                className="card"
                style={
                  Object {
                    "width": "15rem",
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <img
                    className="card-img-top"
                    src="college_logo.png"
                  />
                  <a
                    href="https://collegescorecard.ed.gov/data/"
                  >
                    <div
                      className="card-title h5"
                    >
                      College Score Card API
                    </div>
                  </a>
                  <p
                    className="card-text"
                  >
                    We used the College Score Card API to display information for each of the university instances.
                  </p>
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  <div
    className="project_links"
  >
    <h3>
      Project Links
    </h3>
    <a
      className="btn btn-dark"
      disabled={false}
      href="https://gitlab.com/jakem02/cs373-idb"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      GitLab Project Repository
    </a>
                
    <a
      className="btn btn-dark"
      disabled={false}
      href="https://documenter.getpostman.com/view/23653833/2s83tGoBu1"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Postman API Documentation
    </a>
  </div>,
]
`;

exports[`Renders Coffee Shops page 1`] = `
<div
  className="list"
>
  <h1
    className="pageName"
  >
    Coffee Shops
  </h1>
  <table
    className="table table-striped table-bordered table-hover"
    id="entries"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          City
        </th>
        <th>
          Price
        </th>
        <th>
          Rating
        </th>
        <th>
          Open/Closed
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <div
    className="vstack minBreakpoint-xs"
  >
    <div
      className="model_page_pagination"
    >
      <ul
        className="pagination"
      >
        <li
          className="page-item disabled"
        >
          <span
            className="page-link"
            disabled={true}
            onClick={[Function]}
          >
            <span
              aria-hidden="true"
            >
              ‹
            </span>
            <span
              className="visually-hidden"
            >
              Previous
            </span>
          </span>
        </li>
        <li
          className="page-item active"
        >
          <span
            className="page-link"
            disabled={false}
            onClick={[Function]}
          >
            1
            <span
              className="visually-hidden"
            >
              (current)
            </span>
          </span>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            2
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            3
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            4
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            5
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              …
            </span>
            <span
              className="visually-hidden"
            >
              More
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            105
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            106
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            107
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            108
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            109
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              ›
            </span>
            <span
              className="visually-hidden"
            >
              Next
            </span>
          </a>
        </li>
      </ul>
    </div>
    <div>
      Showing 
      0
       items
    </div>
  </div>
</div>
`;

exports[`Renders Libraries page 1`] = `
<div
  className="list"
>
  <h1
    className="pageName"
  >
    Libraries
  </h1>
  <table
    className="table table-striped table-bordered table-hover"
    id="entries"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Location
        </th>
        <th>
          Rating
        </th>
        <th>
          Telephone
        </th>
        <th>
          Status
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <div
    className="vstack minBreakpoint-xs"
  >
    <div
      className="model_page_pagination"
    >
      <ul
        className="pagination"
      >
        <li
          className="page-item disabled"
        >
          <span
            className="page-link"
            disabled={true}
            onClick={[Function]}
          >
            <span
              aria-hidden="true"
            >
              ‹
            </span>
            <span
              className="visually-hidden"
            >
              Previous
            </span>
          </span>
        </li>
        <li
          className="page-item active"
        >
          <span
            className="page-link"
            disabled={false}
            onClick={[Function]}
          >
            1
            <span
              className="visually-hidden"
            >
              (current)
            </span>
          </span>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            2
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            3
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            4
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            5
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              …
            </span>
            <span
              className="visually-hidden"
            >
              More
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            244
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            245
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            246
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            247
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            248
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              ›
            </span>
            <span
              className="visually-hidden"
            >
              Next
            </span>
          </a>
        </li>
      </ul>
    </div>
    <div>
      Showing 
      0
       items
    </div>
  </div>
</div>
`;

exports[`Renders Navigation bar 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark"
>
  <div
    className="container"
  >
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rr-ui-event-key="/"
        disabled={false}
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="nav-link"
        data-rr-ui-event-key="/CoffeeShops"
        disabled={false}
        href="/CoffeeShops"
        onClick={[Function]}
      >
        Coffee Shops
      </a>
      <a
        className="nav-link"
        data-rr-ui-event-key="/Libraries"
        disabled={false}
        href="/Libraries"
        onClick={[Function]}
      >
        Libraries
      </a>
      <a
        className="nav-link"
        data-rr-ui-event-key="/Universities"
        disabled={false}
        href="/Universities"
        onClick={[Function]}
      >
        Universities
      </a>
      <a
        className="nav-link"
        data-rr-ui-event-key="/About"
        disabled={false}
        href="/About"
        onClick={[Function]}
      >
        About
      </a>
    </div>
  </div>
</nav>
`;

exports[`Renders Pagination Component 1`] = `
<ul
  className="pagination"
>
  <li
    className="page-item disabled"
  >
    <span
      className="page-link"
      disabled={true}
      onClick={[Function]}
    >
      <span
        aria-hidden="true"
      >
        ‹
      </span>
      <span
        className="visually-hidden"
      >
        Previous
      </span>
    </span>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        aria-hidden="true"
      >
        …
      </span>
      <span
        className="visually-hidden"
      >
        More
      </span>
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        aria-hidden="true"
      >
        ›
      </span>
      <span
        className="visually-hidden"
      >
        Next
      </span>
    </a>
  </li>
</ul>
`;

exports[`Renders Universities page 1`] = `
<div
  className="list"
>
  <h1
    className="pageName"
  >
    Universities
  </h1>
  <table
    className="table table-striped table-bordered table-hover"
    id="entries"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          City
        </th>
        <th>
          Zip
        </th>
        <th>
          Undergraduate Population
        </th>
        <th>
          In State Tuition
        </th>
        <th>
          Out of State Tuition
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <div
    className="vstack minBreakpoint-xs"
  >
    <div
      className="model_page_pagination"
    >
      <ul
        className="pagination"
      >
        <li
          className="page-item disabled"
        >
          <span
            className="page-link"
            disabled={true}
            onClick={[Function]}
          >
            <span
              aria-hidden="true"
            >
              ‹
            </span>
            <span
              className="visually-hidden"
            >
              Previous
            </span>
          </span>
        </li>
        <li
          className="page-item active"
        >
          <span
            className="page-link"
            disabled={false}
            onClick={[Function]}
          >
            1
            <span
              className="visually-hidden"
            >
              (current)
            </span>
          </span>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            2
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            3
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            4
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            5
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              …
            </span>
            <span
              className="visually-hidden"
            >
              More
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            25
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            26
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            27
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            28
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            29
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              ›
            </span>
            <span
              className="visually-hidden"
            >
              Next
            </span>
          </a>
        </li>
      </ul>
    </div>
    <div>
      Showing 
      0
       items
    </div>
  </div>
</div>
`;
